source/MaterialXGenGlsl/Nodes/UnlitSurfaceNodeGlsl.cpp

```diff
@@ -0,0 +1,56 @@
+//
+// TM & (c) 2022 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenGlsl/Nodes/UnlitSurfaceNodeGlsl.h>
+
+#include <MaterialXGenShader/Shader.h>
+#include <MaterialXGenShader/GenContext.h>
+
+MATERIALX_NAMESPACE_BEGIN
+
+ShaderNodeImplPtr UnlitSurfaceNodeGlsl::create()
+{
+    return std::make_shared<UnlitSurfaceNodeGlsl>();
+}
+
+void UnlitSurfaceNodeGlsl::emitFunctionCall(const ShaderNode& node, GenContext& context, ShaderStage& stage) const
+{
+    const GlslShaderGenerator& shadergen = static_cast<const GlslShaderGenerator&>(context.getShaderGenerator());
+
+    BEGIN_SHADER_STAGE(stage, Stage::PIXEL)
+
+        // Declare the output variable
+        const ShaderOutput* output = node.getOutput();
+        shadergen.emitLineBegin(stage);
+        shadergen.emitOutput(output, true, true, context, stage);
+        shadergen.emitLineEnd(stage);
+
+        const string outColor = output->getVariable() + ".color";
+        const string outTransparency = output->getVariable() + ".transparency";
+
+        const ShaderInput* emission = node.getInput("emission");
+        const ShaderInput* emissionColor = node.getInput("emission_color");
+        shadergen.emitLine(outColor + " = " + shadergen.getUpstreamResult(emission, context) + " * " + shadergen.getUpstreamResult(emissionColor, context), stage);
+        
+        if (context.getOptions().hwTransparency)
+        {
+            const ShaderInput* transmission = node.getInput("transmission");
+            const ShaderInput* transmissionColor = node.getInput("transmission_color");
+            shadergen.emitLine(outTransparency + " = " + shadergen.getUpstreamResult(transmission, context) + " * " + shadergen.getUpstreamResult(transmissionColor, context), stage);
+
+            const ShaderInput* opacity = node.getInput("opacity");
+            const string surfaceOpacity = shadergen.getUpstreamResult(opacity, context);
+            shadergen.emitLine(outColor + " *= " + surfaceOpacity, stage);
+            shadergen.emitLine(outTransparency + " = mix(vec3(1.0), " + outTransparency + ", " + surfaceOpacity + ")", stage);
+        }
+        else
+        {
+            shadergen.emitLine(outTransparency + " = vec3(0.0)", stage);
```

Comment:
Shouldn't this be 1.0 (meaning fully opaque) otherwise if blending is enabled this will not show up ?

Reply:
Transparency is the complement of opacity, so 0.0 transparency means fully opaque :)